upgrade_common_services.sh

```diff
@@ -0,0 +1,388 @@
+      -cloudpaksNS                  specify the namespace where cloud paks is installed. By default it would be same as csNS.
+      -controlNS                    specify the namespace where singleton services are installed. By default it it would be same as csNS.
+      -c                            specify the subscription channel where common services switch. By default it is channel v3
+	Note:
+	If there is no namespace defined through arguments, all Common Service in the cluster will be upgraded.
+EOF
+}
+
+function main() {
+    CS_NAMESPACE=${CS_NAMESPACE:-ibm-common-services}
+    DESTINATION_CHANNEL=${DESTINATION_CHANNEL:-v3}
+    ALL_NAMESPACE=${ALL_NAMESPACE:-true}
+    
+    while [ "$#" -gt "0" ]
+    do
+        case "$1" in
+        "-h"|"--help")
+            usage
+            exit 0
+            ;;
+        "-csNS")
+            CS_NAMESPACE=$2
+            ALL_NAMESPACE="false"
+            shift
+            ;;
+        "-cloudpaksNS")
+            cloudpaksNS=$2
+            ALL_NAMESPACE="false"
+            shift
+            ;;
+        "-controlNS")
+            controlNS=$2
+            ALL_NAMESPACE="false"
+            shift
+            ;;
+        "-c")
+            DESTINATION_CHANNEL=$2
+            shift
+            ;;
+        *)
+            warning "invalid option -- \`$1\`"
+            usage
+            exit 1
+            ;;
+        esac
+        shift
+    done
+
+    CLOUDPAKS_NAMESPACE=${CLOUDPAKS_NAMESPACE:-${CS_NAMESPACE}}
+    CONTROL_NAMESPAVE=${CONTROL_NAMESPAVE:-${CS_NAMESPACE}}
+
+    if [[ "${ALL_NAMESPACE}" == "true" ]]; then
+        title "Upgrade Commmon Service Operator in all namespaces."
+    else
+        title "Upgrade Common Service Operator to ${DESTINATION_CHANNEL} channel in ${CS_NAMESPACE} namespace."
+    fi
+    msg "-----------------------------------------------------------------------"
+
+    check_preqreqs "${CS_NAMESPACE}" "${CLOUDPAKS_NAMESPACE}" "${CONTROL_NAMESPAVE}"
+    switch_channel "${CS_NAMESPACE}" "${CLOUDPAKS_NAMESPACE}" "${CONTROL_NAMESPAVE}" "${DESTINATION_CHANNEL}" "${ALL_NAMESPACE}"
+    check_switch_complete "${CS_NAMESPACE}" "${CLOUDPAKS_NAMESPACE}" "${CONTROL_NAMESPAVE}" "${DESTINATION_CHANNEL}" "${ALL_NAMESPACE}"
+}
+
+
+function check_preqreqs() {
+    local csNS=$1
+    local cloudpaksNS=$2
+    local controlNS=$3
+    title "[${STEP}] Checking prerequesites ..."
+    msg "-----------------------------------------------------------------------"
+
+    # checking oc command
+    if [[ -z "$(command -v oc 2> /dev/null)" ]]; then
+        error "OpenShift Command Line tool oc is not available"
+    else
+        success "OpenShift Command Line tool oc is available."
+    fi
+
+    # checking oc command logged in
+    user=$(oc whoami 2> /dev/null)
+    if [ $? -ne 0 ]; then
+        error "You must be logged into the OpenShift Cluster from the oc command line."
+    else
+        success "oc command logged in as ${user}"
```

Comment:
Do we need a specific kind of user for this script? Would the user need to be a cluster admin to run upgrades?